actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -637,16 +637,22 @@ def normalize_action_path(action_path) # :doc:
         uri = URI.parse(action_path)
 
         if uri.relative? && (action_path.blank? || !action_path.start_with?("/"))
-          uri = URI.parse(request.path)
-          # add the action path to the request.path
-          uri.path += "/#{action_path}"
-          # relative path with "./path"
-          uri.path.gsub!("/./", "/")
+          return normalize_relative_action_path(uri.path)
```

Comment:
We avoid early return. If there are now two branch of code, make those branch explicit instead of returning early.

```suggestion
        if uri.relative? && (action_path.blank? || !action_path.start_with?("/"))
          normalize_relative_action_path(uri.path)
        else
          uri.path.chomp("/")
        end
```

Reply:
Done.